File: Pattern of Strings/Pattern_of_String.java
class Solution {
    ArrayList<String> pattern(String s) {
        // here we have to print character in each row
	    // start with full length and decrease by 1
	    int n = s.length();
	    ArrayList<String> ans = new ArrayList<>();
	    for(int i = n; i>0; i--) {
	        String temp = "";
	        for(int j =0; j<i; j++) {
	           temp += s.charAt(j);
	        }
	        ans.add(temp);
	    }
	    return ans;
    }
};

File: Pattern of Strings/Pattern_of_String.cpp
class Solution{
public:	
	
	vector<string> pattern(string s)
	{
	    // here we have to print character in each row
	    // start with full length and decrease by 1
	    int n = s.size();
	    vector<string> ans;
	    for(int i = n; i>0; i--) {
	        string temp = "";
	        for(int j =0; j<i; j++) {
	           temp += s[j];
	        }
	        ans.push_back(temp);
	    }
	    return ans;
	}

};